src/app/shared/components/file-upload/file-upload.component.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  forwardRef,
+  Input,
+  Output
+} from '@angular/core'
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms'
+import { ValidationMessageComponent } from '@shared/components/validation-message/validation-message.component'
+import { FormLabelDirective } from '@shared/directives/form-label/form-label.directive'
+import { NgIf } from '@angular/common'
+
+@Component({
+  selector: 'file-upload',
+  standalone: true,
+  imports: [ValidationMessageComponent, FormLabelDirective, NgIf],
+  templateUrl: './file-upload.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => FileUploadComponent),
+      multi: true
+    }
+  ]
+})
+export class FileUploadComponent implements ControlValueAccessor {
+  @Input() public allowedExtensions: string[] = ['.jpg', '.jpeg']
+  @Input() public formControl: FormControl
+  @Input() public url: string | ArrayBuffer = null
```

Comment:
Consider if this is even needed. Maybe we can reuse form control value to store both string and File?